### PR TITLE
SCVS-681 Save to loanrecoverysettle

### PR DESCRIPTION
**Pull request by patilshivani**
Lambda to save to loanrecoverysettle table in agna

### CODE REVIEW

File: SaveToLoanRecoverySettleTable.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    u_id = str(uuid.uuid4())
+    handler = LoanRecoverySettle(appContext, appContext.getEventTime())
+    handler.handle(data)
+    dt = handler.ctx.getObj('settleRecord')
+    datastr = dt.toJSON().encode('utf-8')
+    headers = {'content-type': 'application/json'}
+    r = requests.post(appContext.getAgnaURL(), data=datastr, headers=headers)
+
+    if r.status_code >= 300:
+        logger.error("requestId={0},urlstr= {2},data={3} "
```
Comment (gkaraleuplift) [inline SaveToLoanRecoverySettleTable.py]: Could you please put searchable phrase in logger.error ? I believe that will help in searching log in sumo

File: SaveToLoanRecoverySettleTable.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        return constructResponse(400, 'Error {} in processing. Make sure '
+                                      'all required information is passed'.
+                                 format(a))
+
+    return constructResponse(201, 'FFAM record saved to the table')
+
+
+def updateRecordInAgna(appContext, data):
+    if data is None:
+        return
```
Comment (gkaraleuplift) [inline SaveToLoanRecoverySettleTable.py]: Could you please logger here ?

Comment (patilshivani) [reply SaveToLoanRecoverySettleTable.py]: Incorporated

File: ffammain.py [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
-
+from datetime import datetime
 
+import SaveToLoanRecoverySettleTable
 
 # Below function can be use to test Suspense NLS Loan lambda locally
 # 1) Rename the function to "--main__"
 # 2) Start the Agna project locally
 # 3) Run the script
 if __name__ == "__main__":
```
Comment (gkaraleuplift) [inline ffammain.py]: Put separate main function for this lambda

**Comment by patilshivani** [comment]
> Please add sam template for this lambda

Added

File: ffammain.py (hunk 3 above)
Comment (patilshivani) [reply ffammain.py]: Incorporated

File: samTemplate.yaml [hunk 4]
```diff
@@ -90,3 +90,31 @@ Resources:
...
           - Fn::ImportValue:
               !Sub "${DeploymentStackName}:PrivateSubnetTwo"
+
+  UpdateLoanRecoverySettleFn:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      MemorySize: 512
+      Timeout: 300
+      Handler: SaveToLoanRecoverySettleTable.lambda_handler
+      Runtime: python2.7
```
Comment (gkaraleuplift) [inline samTemplate.yaml]: Could you please use python 3.7 ?

File: ffammain.py [hunk 5]
```diff
@@ -1,27 +1,45 @@
...
         "recordLog": "Hello Testing it Lambda",
-        "timestamp": "2019-06-03T20:23:40.736Z"
+        "timestamp": "2019-06-13T20:23:40.736Z"
     }
     event['time'] = datetime.utcnow()
-    SaveToLoanRecoveryRecordSentTable.lambda_handler(event,None)
+    SaveToLoanRecoveryRecordSentTable.lambda_handler(event, None)
+
+
+if __name__ == "__main__":
```
Comment (gkaraleuplift) [inline ffammain.py]: rename __main__ to __main__1

**Pull request merged**